code/datums/components/uplink.dm

```diff
@@ -82,8 +82,17 @@
 	purchase_log = null
 	return ..()
 
-/datum/component/uplink/proc/update_items()
-	uplink_items = get_uplink_items(gamemode, TRUE, allow_restricted)
+/datum/component/uplink/proc/update_items(update_sales = FALSE)
```

Comment:
This is a bad boolean parameter, I think you should instead just handle updating discounted gear in its own proc.